src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,100 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final float GROW_FACTOR = 1.5F;
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] values;
+
+    public ArrayList() {
+        values = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= values.length) {
+            values = grow();
+        }
```

Comment:
```suggestion
        growIfNeeded();
```
let's move this logic to the method